### DISCUSSION TITLE
Code Hangs during the implementation of DDP

### DISCUSSION THREAD

**Discussion by Mukil07** [optuna/optuna#5918] [Q&A]
The code works perfectly fine for single gpu and also with multigpu (DDP). But while using optuna, the code hangs until the GPU processes gets killed at this line `model = DDP(model)`. 

I have followed the same procedure followed by this example 
https://github.com/optuna/optuna-examples/blob/main/pytorch/pytorch_distributed_simple.py

Im trying to tune only the learning rate and not the model. Also using custom dataloader, and train loops.

**Comment by nzw0301** [comment]
Could you share the [minimal reproducible code](https://stackoverflow.com/help/minimal-reproducible-example)? Otherwise, it's really hard to answer your question.
